crates/transaction-pool/src/traits.rs

```diff
@@ -753,6 +753,33 @@ impl fmt::Display for CanonicalStateUpdate<'_> {
     }
 }
 
+/// Helper error type representing different reasons why a transaction might fail during
+/// processing in the transaction pool.
+#[derive(Debug, derive_more::Display, derive_more::Error)]
+pub enum InvalidKind {
```

Comment:
I think there might be other reasons that aren't necessarily consensus-related, so I think we can also add an `Other` variant here

Reply:
`InvalidPoolTransactionError` has already an `Other`variant, yet I thought 
```rust    
InvalidPoolTransactionError::Consensus(InvalidTransactionError::TxTypeNotSupported),
``` could be adapted